### ISSUE TITLE
Items overlaps on push directions to east and west only

### ISSUE THREAD

**Issue by anjmao** [tiberiuzuld/angular-gridster2#177]
Hi @tiberiuzuld, I'm trying to fix one case when I have push set to only east as west.

Configuration
```js
vm.options = {
        gridType: 'fit',
        compactType: 'compactUp&Left',
        margin: 10,

        minCols: 6,
        maxCols: 6,
        minRows: 4,
        maxRows: 4,

        draggable: {
          delayStart: 0,
          enabled: true,
          stop: eventStop
        },
        resizable: {
          delayStart: 0,
          enabled: true,
          stop: eventStop
        },
        
        swap: false,
        pushItems: true,
        pushDirections: { north: false, east: true, south: false, west: true },
        displayGrid: 'onDrag&Resize',
        disableWindowResize: false
      };
```

Dashboard items
```js
vm.dashboard = [

        { cols: 2, rows: 2, x: 4, y: 0 },
        { cols: 2, rows: 2, x: 0, y: 0 },

        { cols: 1, rows: 1, x: 2, y: 0 },
        { cols: 1, rows: 1, x: 3, y: 0 },
        { cols: 1, rows: 1, x: 2, y: 1 },
        { cols: 1, rows: 1, x: 3, y: 1 },
      ]
```
Problem:
If I move `Hello 4` to `Hello 2`, `Hello 2` will overlay with `Hello 0` at the right, but in this case it should not allow to change items positions.
<img width="1435" alt="case" src="https://user-images.githubusercontent.com/3028012/33605699-74c56ed0-d9c3-11e7-8f5d-4942ff15c3a8.png">

**Comment by tiberiuzuld** [comment]
Hi @anjmao ,
Will have a look over it this week.

Thanks

**Comment by tiberiuzuld** [comment]
Hi @anjmao ,
Fixed in 4.2.0
Let me know if you encounter any more issues.

Thanks
